tcms/telemetry/static/telemetry/js/testing/status-matrix.js

```diff
@@ -0,0 +1,117 @@
+$(document).ready(() => {
+
+    loadInitialProduct();
+
+    document.getElementById('id_product').onchange = () => {
+        update_version_select_from_product();
+        update_build_select_from_product();
+        updateTestPlanSelectFromProduct();
+
+        redrawTable();
+    };
+
+    document.getElementById('id_version').onchange = redrawTable;
+    document.getElementById('id_build').onchange = redrawTable;
+    document.getElementById('id_test_plan').onchange = redrawTable;
+
+    $('#id_after').on('dp.change', redrawTable);
+    $('#id_before').on('dp.change', redrawTable);
+
+    drawTable();
+});
+
+function drawTable(query = {}) {
```

Comment:
IMO we should use `dataTableJsonRPC()` to feed the results. The code should not care about appending table cell or manipulating the HTML structure of the table.

There would be some rendering functions some of which can be pulled from the search pages and reused, e.g. test case name should be a link, possibly with `TC-XX:` prefix.